src/correlation.rs

```diff
@@ -198,6 +198,30 @@ impl Correlations {
 
         Ok(())
     }
+
+    pub async fn delete_correlations_for_user(
+        &self,
+        user_id: &str,
+        session_key: &SessionKey,
+    ) -> Result<(), CorrelationError> {
+        let user_correlations =
+            CORRELATIONS
```

Comment:
```suggestion
            self
```

We can directly mutate correlations state from her and not have to call `list/delete`

Reply:
` fn list_correlations` was checking some permissions for user, not sure what they are.. so i tried to include it. Will implement as you suggested.